### PR TITLE
Add gnuradio 3.8, gr-iio 3.8, gr-m2k, gr-scopy, libm2k as dependencies

### PR DESCRIPTION
**Pull request by adisuciu**
Added dependencies to CMake, and updated build system.

### CODE REVIEW

**Comment by matejk** [comment]
Automated build with all dependencies and creating deb packages was implemented a bit more than a year ago and is available on branch:

https://github.com/matejk/scopy/tree/uberscope-dev

Perhaps something can be useful to include mainstream.

File: CMakeLists.txt [hunk 1]
```diff
@@ -20,6 +20,8 @@
 cmake_minimum_required(VERSION 3.5)
 project(scopy LANGUAGES CXX VERSION 1.1.2)
 
+message(${CMAKE_SYSTEM_PREFIX_PATH})
```
Comment (DanielGuramulta) [inline CMakeLists.txt]: Is this message still needed?

File: appveyor.yml [hunk 2]
```diff
@@ -4,100 +4,40 @@ clone_depth: 1
...
-    - mv C:\%DEST_FOLDER%\.debug C:\%DEBUG_FOLDER%
-
-    - 7z a "c:\scopy-%ARCH_BIT%bit.zip" c:\%DEST_FOLDER%
-    - appveyor PushArtifact c:\scopy-%ARCH_BIT%bit.zip
-    - 7z a "C:\debug-%ARCH_BIT%bit.zip" c:\%DEBUG_FOLDER%
-    - appveyor PushArtifact C:\debug-%ARCH_BIT%bit.zip
-    
+    - cmd: set PATH=%PATH%;"C:\Program Files (x86)\Inno Setup 5"
+    - cmd: C:\msys64\usr\bin\bash build_appveyor_mingw.sh
+    - sh: /home/appveyor/projects/scopy/install_deps.sh
```
Comment (DanielGuramulta) [inline appveyor.yml]: I think we can move this files (install_deps.sh, ...) from the top directory into another one just to keep things clean.
The files could be moved in a folder `ci/appveyor/`, we already have `ci/travis`

File: build_appveyor_mingw.sh [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+	http://repo.msys2.org/mingw/$ARCH/mingw-w64-$ARCH-qt5-5.13.2-1-any.pkg.tar.xz \
+	http://repo.msys2.org/mingw/$ARCH/mingw-w64-$ARCH-libusb-1.0.21-2-any.pkg.tar.xz \
+"
+
+DLL_DEPS="libmatio-*.dll libhdf5-*.dll libszip*.dll libpcre*.dll libdouble-conversion*.dll libwinpthread-*.dll libgcc_*.dll libstdc++-*.dll libboost_*.dll libglib-*.dll libintl-*.dll libiconv-*.dll libglibmm-2.*.dll libgmodule-2.*.dll libgobject-2.*.dll libffi-*.dll libsigc-2.*.dll libfftw3f-*.dll libicu*.dll zlib*.dll libharfbuzz-*.dll libfreetype-*.dll libbz2-*.dll libpng16-*.dll libgraphite2.dll libjpeg-*.dll libsqlite3-*.dll libwebp-*.dll libxml2-*.dll liblzma-*.dll libxslt-*.dll libzip*.dll libpython3.*.dll libgnutls*.dll libnettle*.dll libhogweed*.dll libgmp*.dll libidn*.dll libp11*.dll libtasn*.dll libunistring*.dll libusb-*.dll libzstd*.dll libgnuradio-*.dll /$MINGW_VERSION/lib/python3.* libiio*.dll libvolk*.dll liblog4cpp*.dll libad9361*.dll liborc*.dll libsigrok*.dll qwt*.dll"
+
+#do we need this ?
+#http://swdownloads.analog.com/cse/build/windres.exe.gz;\
+
+#if [ ${ARCH} == "i686" ]
```
Comment (DanielGuramulta) [inline build_appveyor_mingw.sh]: If we don't need this commented lines we could simply remove them

**Comment by adisuciu** [comment]
> Automated build with all dependencies and creating deb packages was implemented a bit more than a year ago and is available on branch:
> 
> https://github.com/matejk/scopy/tree/uberscope-dev
> 
> Perhaps something can be useful to include mainstream.

Hello, we'll take a look at your work and see what we can use in mainstream
-Adrian

File: appveyor.yml (hunk 2 above)
Comment (AlexandraTrifan) [reply appveyor.yml]: We will move these files in the next PR (which will contain Appveyor flatpak build and Appveyor macOS build)

File: CMakeLists.txt (hunk 1 above)
Comment (adisuciu) [reply CMakeLists.txt]: fixed

**Pull request merged**